.github/workflows/r.yml

```diff
@@ -327,6 +327,13 @@ jobs:
         shell: Rscript {0}
         working-directory: r
         run: |
+          Sys.setenv(
+            RWINLIB_LOCAL = file.path(Sys.getenv("GITHUB_WORKSPACE"), "r", "windows", "libarrow.zip"),
+            MAKEFLAGS = paste0("-j", parallel::detectCores()),
+            ARROW_R_DEV = TRUE,
+            "_R_CHECK_FORCE_SUGGESTS_" = FALSE
+          )
+          pak::pak("local::.")
```

Comment:
Would `pak::local_install()` work here? https://pak.r-lib.org/reference/local_install.html Or at least a comment about why not | what it's doing?

Reply:
AFAIU `pak::pak()` and `pak::pkg_install()` are equivalent (https://pak.r-lib.org/reference/pak.html#details-1) and so are `pak::local_install()` and `pak::pkg_install("local::.")` (https://pak.r-lib.org/reference/local_install.html#details-1) => `pak::local_install()` and `pak::pak("local::.")` are equivalent.

Are you thinking that `pak::local_install()` is a bit more readable? I would think people modifying this code are somewhat familiar with `pak`, and thus they encountered the `"local::."` incantation. For me, the `local_` pattern is less readable as there is a bit of overlap and different meaning than `withr::local_`. I read `local_` as we're doing something locally, whereas in this context it actually means we are installing the package located in the root directory. From this point of view, I think `"local::."` is less ambiguous.

Reply:
I'm not familiar enough with {pak} and {pak}'s idioms to know which are the better or more natural ones for {pak}. https://pak.r-lib.org/reference/pak_package_sources.html#local-package-trees-1 lists `local_install()` first and it's easier to get to that from the reference section of the docs, which is what I first tried to do when I was trying to figure out what `pak::pak("local::.")` was doing while reviewing this (the docs for `pak::pak()` doesn't list what `"local::."` means, which was the first place I went). 

I don't think that we can be certain people modifying this code will be familiar with {pak} (as far as I know this is the only place in the whole repo we use it). So we should flag what's up with a comment if we're using something that's not easy to get to the documentation about. We might even also add in why {pak} is helpful here over other approaches

Reply:
I switched to `pak::local_install()` as your search experience points to that documentation being a bit more accessible. I also added a comment.